tl.tl

```diff
@@ -3853,9 +3861,23 @@ local function parse_return(ps: ParseState, i: integer): integer, Node
    return i, node
 end
 
-local function store_field_in_record(ps: ParseState, i: integer, field_name: string, newt: Type, fields: {string: Type}, field_order: {string}): boolean
+local function store_field_in_record(ps: ParseState, i: integer, field_name: string, newt: Type, def: RecordLikeType, comments?: {Comment}, meta?: MetaMode): boolean
+   local field_order, fields, field_comments: {string}, {string:Type}, {string: {{Comment}}}
+   if meta then
+      field_order, fields, field_comments = def.meta_field_order, def.meta_fields, def.meta_field_comments
+   else
+      field_order, fields, field_comments = def.field_order, def.fields, def.field_comments
+   end
+
+   if comments and not field_comments then 
+      field_comments = {}
```

Comment:
this new table isn't being attached back into `def`.

I can see that this is working because you're still always creating one below for `def.field_comments` (line 4097), but I don't see an equivalent for `def.meta_field_comments`, which made me wonder how the tests were passing.

But upon a closer look at the tests, I see that the loop in the test was iterating over the wrong table resulting in an empty loop (see the other comment above).